coalib/parsing/Globbing.py

```diff
@@ -29,6 +29,39 @@ def _end_of_set_index(string, start_index):
     return closing_index
 
 
+def escape_brackets_and_parentheses(string):
+    """
+    Escapes the given string with [c] pattern
+    """
+    # We need to handle '[' and ']' specially
+    tmp = ""
+    prev = 0
+    while True:
+        lb = string[prev:].find("[")
+        rb = string[prev:].find("]")
+        if lb is -1 or rb is -1:
+            cur = max(lb, rb)
+        else:
+            cur = min(lb, rb)
+        if cur is -1:
+            tmp += string[prev:]
+            break
+        cur = cur + prev
+        tmp += string[prev:cur]
+        tmp += "[" + string[cur] + "]"
+        prev = cur + 1
+    return tmp.replace("(", "[(]").replace(")", "[)]")
+
+
+def unescape_brackets_and_parentheses(string):
+    """
+    Unescapes the given string
+    Note: Only for strings escaped with escape_brackets_and_parentheses
+    """
+    string = string.replace("[(]", "(").replace("[)]", ")")
+    return string.replace("[[]", "[").replace("[]]", "]")
```

Comment:
@sils1297 If it is named `glob_escape` it would lose its primary meaning - escaping brackets and parentheses. I understand that it uses globs to escape, but I'm not really sure if it should be named as `glob_escape`.